linter.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+import subprocess
+import glob
+
+
+def load_all_json_files(base_path: str) -> list[str]:
+    """
+    Load all json files in the repository.
+
+    Args:
+        base_path: str
+
+    Returns:
+        list[str]:
+    """
+    return list(glob.glob(base_path + '/**/*.json', recursive=True))
+
+
+def find_changed_json_files() -> list[str]:
+    """
+    Find all changed json files in the repository.
+
+    Returns:
+        list[str]:
+    """
+    diff = subprocess.run(
+        ["git", "diff", "--name-only", "origin/main"],
+        capture_output=True
+    )
+
+    return [
+        changed_file.decode()
+        for changed_file in diff.stdout.splitlines()
+        if changed_file.decode().endswith(".json")
+    ]
+
+
+def run_npx_on_files(files_to_fix: list[str], check_only: bool = True):
+    """
+    Run prettier on the list of files.
+
+    Args:
+        files_to_fix: list[str]
+        check_only: bool
+    """
+    action = "--check" if check_only else "--write"
+
+    result = subprocess.run(
+        ["npx", "prettier", action] + files_to_fix,
+        capture_output=True
+    )
+
+    if result.returncode != 0:
+        raise Exception("Prettier failed: \n" + result.stderr.decode())
+    else:
+        print("Prettier run successfully: \n" + result.stdout.decode())
```

Comment:
Can we not rely on the python json module (like [this example](https://github.com/SEKOIA-IO/intake-formats/blob/main/utils/linter.py#L63)) instead of `prettier`? What do you think about? It will avoid contributors to install `prettier`.

Reply:
Use json.dumps instead